### DISCUSSION TITLE
Question about listener.tcp.external

### DISCUSSION THREAD

**Discussion by roysG** [emqx/emqx#7539] [Q&A]
According to this url:
https://www.emqx.io/docs/en/v4.4/tutorial/tune.html#erlang-vm-tuning

For prepare the cluster to production your recommendation is to change the line "listener.tcp.external.acceptors" to "64":
listener.tcp.external.acceptors = 64

In case i  use in ssl connection, do i need to change also the line below to 64?

Current:
listener.ssl.external.acceptors = 16 ? 
New change:
listener.ssl.external.acceptors = 64 ? 

Also please tell us, what is the purpose of this line

Thanks

**Comment by zmstone** [comment]
By increasing the number of acceptors, you can allow more concurrent incoming connections.
SSL tend to be more resource intensive, so it’s suggested a smaller number.

from our experience though, the acceptors are usually not the bottleneck, especially when external auth integration is in use.
In order not to overwhelm the backends, quite many of our users have to lower the conn_rate.

**Reply by roysG** [reply]
Is there any best practice for ssl, like you wrote for the regular protocol?

**Reply by zmstone** [reply]
Really depends on your requirements and environment.

Without more information, I’d say go with 16.
Should be fast enough for most of the cases.

**Comment by roysG** [comment]
Ok thanks

**Comment by roysG** [comment]
The configuration for production also good for ssl protocl, right?
